05/app.js

```diff
@@ -1,53 +1,75 @@
 const stats = {
     paragraphs: {
-        'p1': 0,
+        p1: 0,
     },
     links: {
-        '/dolor.html': 0,
-    }
+        "/dolor.html": 0,
+    },
 };
+const pEl = document.querySelectorAll("p");
+pEl.forEach(function (e) {
+    e.addEventListener("click", function (e) {
+        e.preventDefault();
+        console.log("click", e.target);
 
+        if (e.target.tagName === "P") {
+            const id = e.target.dataset.id;
+            if (typeof stats.paragraphs[id] === "undefined") {
+                stats.paragraphs[id] = 0;
+            }
+            stats.paragraphs[id]++;
+        } else if (e.target.tagName === "A") {
+            const href = e.target.getAttribute("href");
+            if (typeof stats.links[href] === "undefined") {
+                stats.links[href] = 0;
+            }
+            stats.links[href]++;
+        }
+    });
+});
```

Comment:
Super sobie poradziłeś!

Teraz można zastanowić się jak to zrefaktoryzować (sprawnić).
Można zauważyć że w liniach 16-20 oraz 22-27 dzieją się podobne rzeczy.
Może da się napisać funkcję, która będzie przyjmować informacje o nazwie elementu (links lub paragraphs) oraz wartość klucza (zmienna href lub id) i robić co do niej należy? Potraktuj to jako zadanie dodatkowe (niewymagane).